### ISSUE TITLE
Value converter is not working in Data grid.

### ISSUE THREAD

**Issue by john-li-zenith** [akgulebubekir/Maui.DataGrid#214]
.NET 9.0 and DataGrid 4.0.5
I have a simple value converter to return tenant name. However, it is not working in the DataGrid. I took the binding format from the example. Is there something i am missing? 

```
<dg:DataGridColumn
    Title="Tenant"
    Width="0.50*"
    IsVisible="True"
    PropertyName="TenantId">
    <dg:DataGridColumn.CellTemplate>
        <DataTemplate x:DataType="m:Income">
            <ContentView HorizontalOptions="Fill" VerticalOptions="Fill">
                <Label
                    HorizontalOptions="Center"
                    Text="{Binding Converter={StaticResource TenantConverter}}"
                    TextColor="Black"
                    VerticalOptions="Center" />
            </ContentView>
        </DataTemplate>
    </dg:DataGridColumn.CellTemplate>
</dg:DataGridColumn>
```

**Comment by symbiogenesis** [comment]
Have you tested that the converter itself returns the expected result? There's no code to go on, and no expected vs actual value to review.

**Comment by john-li-zenith** [comment]
Yes. the same converter is used in other views and it works okay. I also notice two things, 1) during debug with hot reload, if I delete the label, paste the same label and save, it somehow triggers the datagrid to "recognize" the converter, and all of sudden, it all work. This is very strange. I later confirmed the converter was never called when i first started the app in emulator.  2) I recently updated datagrid to 4.0.5. I forgot what version i used before but when i first developed that datagrid view using the older version, everything was fine.

**Comment by john-li-zenith** [comment]
![Screenshot 2024-12-14 125100](https://github.com/user-attachments/assets/9034be58-039f-4039-9e84-6525379a70bd)
![Screenshot 2024-12-14 124749](https://github.com/user-attachments/assets/a1db5de2-9b73-4df6-bee2-4abf2dfee32a)
Attached are the two screen shots. The converter method is definitely correct. The hot reload can trigger the tenant name to be populated.

**Comment by symbiogenesis** [comment]
There are some changes with respect to AOT with newer MAUI versions.

Different flags were added like: `<MauiEnableXamlCBindingWithSourceCompilation>` and `<PublishAot>`

Have you turned on AOT? You could try turning it off. Is there a difference between Release and Debug mode? Does it happen on all platforms?

Switching from Static to Dynamic resources can also affect things.

Those are just some thoughts off the top of my head. But, if you can make a small reproducible test case, then you might report it upstream.

**Comment by john-li-zenith** [comment]
the converter works outside of the DataGrid, i.e. in a regular CollectionView. I also tested on ios simulator, and got the same issue. Tried debug vs release, same thing. I don't know about AOT, but I will take a look. My gut feeling is the converter has no problem because i can get it work using hot reload. It seems to be some hidden compatibility issue between current version DataGrid and .NET Maui 9?

**Comment by symbiogenesis** [comment]
Try switching to the main branch code from this repo, instead of the nuget package. There were various .NET 9 fixes.

**Comment by symbiogenesis** [comment]
Were you able to solve this one?

**Comment by symbiogenesis** [comment]
I think the latest version will not have any issues
